gravitee-am-common/src/main/java/io/gravitee/am/common/i18n/FreemarkerMessageResolver.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.am.common.i18n;
+
+import freemarker.template.TemplateMethodModelEx;
+import freemarker.template.TemplateModelException;
+
+import java.util.List;
+import java.util.Properties;
+
+import static java.text.MessageFormat.format;
+import static java.util.Optional.ofNullable;
+
+/**
+ * @author Eric LELEU (eric.leleu at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class FreemarkerMessageResolver implements TemplateMethodModelEx {
+    public static final String METHOD_NAME = "msg";
+    private final Properties messages;
+
+    public FreemarkerMessageResolver(Properties messages) {
+        this.messages = messages;
+    }
+
+    @Override
+    public Object exec(List list) throws TemplateModelException {
```

Comment:
Type parameter for `List`? Not sure if this will work, we can ignore it otherwise:
```suggestion
    public Object exec(List<? extends TemplateModel> list) throws TemplateModelException {
```

Reply:
Can't apply it as the TemplateMethodModelEx interface doesn't define generics